suite-native/message-system/src/components/ContextMessage.tsx

```diff
@@ -0,0 +1,170 @@
+
+type ContextMessageProps = {
+    messageId: string;
+};
+
+const CONTEXT_MESSAGE_ANIMATION_DURATION = 250;
+
+type ContextMessageStyle = {
+    backgroundColor: Color;
+    icon: IconName;
+    iconColor: Color;
+    iconBackgroundColor: Color;
+};
+
+const allMessageCategories: Category[] = ['banner', 'context', 'modal', 'feature'];
+
+const ContextMessageVariantToStyleMap = {
+    info: {
+        backgroundColor: 'backgroundAlertBlueSubtleOnElevation0',
+        icon: 'info',
+        iconColor: 'iconAlertBlue',
+        iconBackgroundColor: 'backgroundAlertBlueSubtleOnElevation1',
+    },
+    warning: {
+        backgroundColor: 'backgroundAlertYellowSubtleOnElevation0',
+        icon: 'warningTriangle',
+        iconColor: 'iconAlertYellow',
+        iconBackgroundColor: 'backgroundAlertYellowSubtleOnElevation1',
+    },
+    critical: {
+        backgroundColor: 'backgroundAlertRedSubtleOnElevation0',
+        icon: 'warningOctagon',
+        iconColor: 'iconAlertRed',
+        iconBackgroundColor: 'backgroundAlertRedSubtleOnElevation1',
+    },
+} as const satisfies Record<Variant, ContextMessageStyle>;
+
+const messageContainerStyle = prepareNativeStyle<{ backgroundColor: Color }>(
+    (utils, { backgroundColor }) => ({
+        backgroundColor: utils.colors[backgroundColor],
+        paddingVertical: utils.spacings.medium,
+        paddingLeft: utils.spacings.small / 2,
+        paddingRight: utils.spacings.medium,
+        borderRadius: utils.borders.radii.medium,
+    }),
+);
+
+const messageContentStyle = prepareNativeStyle(() => ({
+    flexDirection: 'row',
+    flexShrink: 1,
+}));
+
+const IconContainerStyle = prepareNativeStyle(utils => ({
+    padding: utils.spacings.small * 1.5,
+    borderRadius: utils.borders.radii.round,
+    justifyContent: 'center',
+    alignItems: 'center',
+}));
+
+const messageTextContainerStyle = prepareNativeStyle(() => ({
+    flexShrink: 1,
+    alignItems: 'flex-start',
+}));
+
+const MessageCloseButton = ({
+    backgroundColor,
+    onClose,
+}: {
+    backgroundColor: Color;
+    onClose: () => void;
+}) => (
+    <TouchableOpacity onPress={onClose}>
+        <RoundedIcon
+            name="close"
+            iconSize="medium"
+            containerSize={44}
+            backgroundColor={backgroundColor}
+        />
+    </TouchableOpacity>
+);
+
+export const ContextMessage = ({ messageId }: ContextMessageProps) => {
+    const dispatch = useDispatch();
+    const { applyStyle } = useNativeStyles();
+
+    const message = useSelector((state: MessageSystemRootState) =>
+        selectMessageById(state, messageId),
+    );
+
+    if (G.isNull(message)) return null;
+
+    // We use only English locale in suite-native so far. When the localization to other
+    // languages is implemented, the language selection logic has to be added here.
+    const messageContent = message.content.en;
+
+    const messageLinkLabel = message.cta?.label.en;
+    const messageLink = message.cta?.link;
```

Comment:
There are two types of links, internal and external

Reply:
Good point. We have decided to not support "internal links" so far in the mobile version. But I at least added a check for `external-link` in fixup https://github.com/trezor/trezor-suite/pull/9105/commits/1fc0085db992732ed544631274226788618721b2.